docs/source/usage/graders/writing_graders.rst

```diff
@@ -0,0 +1,175 @@
+- ``--read <filepath>``: Give the submsission read only access to the specified file.
+
+Note that in both cases, ``filepath`` must be an absolute path.
+
+See :doc:`the file_io example <examples/file_io>` for an example grader utilizing this feature.
+
+.. tip::
+
+   You can use the special argument ``--`` to denote the wrapper
+   should stop parsing arguments and pass the rest of the arguments to the submission.
+   For example::
+
+      submission --write /path/to/file -- arg1 arg2
+
+   will give the submission read/write access to ``/path/to/file``, and pass
+   ``arg1`` and ``arg2`` to the submission.
+
+If you need to upload specific read-only files, please :doc:`contact us </contact>`.
+
+------------
+Grader Files
+------------
+To prevent conflicts/overwriting of other files, all graders should follow the rules below:
+
+Graders should only write to files in the same directory as the grader (i.e. ``Path(__file__).parent``), and the directory
+containing the student submission (i.e. ``Path(sys.argv[2]).parent``).
+
+Do NOT create a file in the grader script directory with the same name as a students username.
+
+Do NOT prefix the name of any files written/read to with ``grader`` - these are reserved for the Tin server itself.
+
+Additionally, since all of a student's submissions are placed in the same directory, files created in the submission directory
+(for example, filenames passed to the submission as output files) should be given random names to avoid
+conflicts in case the student uploads a second submission while their last submission has not yet been graded.
+
+
+-------------
+Grader Output
+-------------
+Students can only see output from the grader that has been printed on the standard output (:obj:`sys.stdout`).
+For example, students would be able to see this::
+
+  print("HEY YOU, STOP CHEATING!")
+
+However, students cannot see anything on :obj:`sys.stderr` - This is to prevent students from
+seeing a solution in the output if the grader throws an exception. For example, only teachers
+would be able to see the following exception::
+
+  raise RuntimeError("Student said 1+1=3")
+
+If the grader script exits with a non-zero status code (which Python does by default when an
+exception is raised) the student will see the text [Grader error] at the end of the output.
+If the grader exceeds its timeout (as set in the assignment "Edit" page), the student will see the text
+[Grader timed out]. Similar text will also be added to the error output.
+
+~~~~~~~~~~~~~~~~~
+Automatic Scoring
+~~~~~~~~~~~~~~~~~
+Each submission has a "Score" field that can be set by the grader. If this field is set,
+you will be able to see a list of each student's scores on the assignment's page,
+which is designed to make entering grades into the gradebook easier.
+
+To set this field, simply print ``Source: <score>`` at the very end, to :obj:`sys.stdout`. For example::
```

Comment:
Typo?

Reply:
Are you talking about the `::`? If so, it's not a type. It renders as a `:` followed by a code block.

Reply:
No... shouldn't it be `Score: <score>` and not `Source: <score>`?

Reply:
Oh I don't know how I missed that lol
I'll fix it, nice catch :)